src/ledger/LedgerTxn.h

```diff
@@ -174,6 +174,27 @@ namespace stellar
 // size, just fail to batch quite as evenly.
 static const size_t LEDGER_ENTRY_BATCH_COMMIT_SIZE = 0xfff;
 
+// If a LedgerTxn has had an eraseWithoutLoading call, the usual "exact"
+// level of consistency that a LedgerTxn maintains with the database will
+// be very slightly weakened: one or more "erase" events may be in
+// memory that would normally (in the "loading" case) have been annihilated
+// on contact with an in-memory insert.
+//
+// This "extra deletes" inconsistency is mostly harmless, it only has two
+// effects:
+//
+//    - LedgerTxnDeltas should not be calculated from a LedgerTxn in
+//      this state (since it will be incorrect!)
+//
+//    - The count of rows in the database effected when applying the
+//      "erase" events might not be the expected number, so the consistency
+//      check we do there should be relaxed.
```

Comment:
Mm. I initially thought this was not going to be a thing because an update or an insert effects the same number of rows at the db level. But it _does_ cause a similar problem in calculating LedgerChanges and LedgerTxnDeltas! Updated accordingly.

(Note: in fixing this, I did _not_ expand the consistency enum to a larger lattice of possible inconsistencies and their combinations; merely generalized/renamed the extra-deletes case to "inexact". While at first I thought this might mean we'd be weakening the rows-effected database-delete consistency check in some cases unnecessarily, thinking it over I think this is not an issue: such a weakened check would only be _unnecessary_ in cases where we were doing a bulk-upsert and _not_ a bulk-delete, but if that were the case then there'd _be_ no bulk-deletes to be weakening. IOW I think the only case we'd weaken _unnecessarily_ would, by definition, not be happening. Also it's somewhat of a corner case anyways, and there are a few ways it's already an overapproximation, not that it's ever very convincing to argue "I'm already being bad, why not be worse?", but TL;DR I don't think this is _actually_ any worse.)